### PR TITLE
Adding timeout for Longevity app to connect.

### PR DESCRIPTION
**Pull request by rogermichoud**
## Overview
Ensures that if the app runtime is not able to connect after some timeout (1 minute default), the app is terminated.

Why should this be merged: 
Sometimes, there is some issue during the run that makes the runtime unable to connect. This could be because of infra problem in the test, but this could also be because of type of faults.

## Checklist (Definition of Done):

- [X] There are no TODOs left in the code
- [X] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [] Change is covered by automated tests
- [X] Public API has Javadoc

### CODE REVIEW

File: generator/src/main/java/org/corfudb/generator/LongevityApp.java [hunk 1]
```diff
@@ -52,7 +55,16 @@ public LongevityApp(long durationMs, int numberThreads, String configurationStri
         this.numberThreads = numberThreads;
 
         operationQueue = new ArrayBlockingQueue<>(QUEUE_CAPACITY);
-        rt = new CorfuRuntime(configurationString).connect();
+
+        rt = new CorfuRuntime(configurationString);
```
Comment (corfu-repo) [inline generator/src/main/java/org/corfudb/generator/LongevityApp.java]: MINOR Remove this use of "CorfuRuntime"; it is deprecated. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

**Comment by corfu-repo** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: generator/src/main/java/org/corfudb/generator/LongevityApp.java [hunk 2]
```diff
@@ -52,7 +55,16 @@ public LongevityApp(long durationMs, int numberThreads, String configurationStri
...
+
+        try {
+            tryToConnectTimeout(TIME_TO_WAIT_FOR_RUNTIME_TO_CONNECT);
```
Comment (Maithem) [inline generator/src/main/java/org/corfudb/generator/LongevityApp.java]: why not inline and do it synchronously ?

**Comment by Maithem** [comment]
@annym is testing blocked by this, can does it need to be merged right away?

**Comment by corfudb-performance** [comment]
Results automatically generated by _CorfuDB Benchmark Framework_ to assess the performance of this pull request for commit **3277903**.

*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 10 threads, Disk mode
 
 An interactive dashboard with Pull Request Performance Metrics for ALL cluster types and numbers of threads in run, is available at:
 [Pull Request #1205 Graphs ](https://snapshot.raintank.io/dashboard/snapshot/ycK1HATHm0pKvUWU4FqfJs0Uw25STApI)

**Comment by annym** [comment]
@no2chem @Maithem can we get this in? it is important for the test framework.

**Comment by corfudb-performance** [comment]
Results automatically generated by _CorfuDB Benchmark Framework_ to assess the performance of this pull request for commit **70b93db**.

*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 10 threads, Disk mode
 
 An interactive dashboard with Pull Request Performance Metrics for ALL cluster types and numbers of threads in run, is available at:
 [Pull Request #1205 Graphs ](https://snapshot.raintank.io/dashboard/snapshot/Ckd7a1uS3j7ow7PG8GeXOI7piXQXQmOe)

**Comment by Maithem** [comment]
@annym  merged.

**Pull request merged**